tools/migrate_to_bzlmod.py

```diff
@@ -0,0 +1,510 @@
+    if s:
+      eprint(line)
+      return s.groups()[0]
+  return None
+
+
+def address_bind_issue(bind_target, resolved_repos):
+  warning(
+      f"A bind target detected: {bind_target}, please fix manually! You should just reference the actual target directory instead of using //external package.")
+
+  name = bind_target.split(":")[1]
+  bind_def = None
+  for dep in resolved_repos:
+    if dep["original_rule_class"] == "bind" and dep["original_attributes"]["name"] == name:
+      bind_def = print_repo_definition(dep)
+      break
+
+  if bind_def:
+    bind_def = ["", "# TODO: Remove the following bind usage"] + bind_def
+    if yes_or_no("Do you wish to add the bind definition to WORKSPACE.bzlmod for later migration?", False):
+      scratch_file("WORKSPACE.bzlmod", bind_def, mode="a")
+      return True
+  else:
+    info(
+        f"Bind definition for {bind_target} isn't found in ./resolved_deps.py file, to get more verbose info please delete "
+        "resolved_deps.py and rerun the script after changing `USE_BAZEL_SYNC` to True in the migration script.")
+
+  return yes_or_no("Do you wish to retry the build after fixes?", True)
+
+
+def grep_file(file_path, pattern):
+  f = pathlib.Path(file_path)
+  if not f.exists():
+    return []
+  result = []
+  with open(f, "r") as i:
+    for line in i:
+      if re.search(pattern, line):
+        result.append(line)
+  return result
+
+
+def _extract_version_number(bazel_version):
+  """Extracts the semantic version number from a version string
+  Args:
+    bazel_version: the version string that begins with the semantic version
+      e.g. "1.2.3rc1 abc1234" where "abc1234" is a commit hash.
+  Returns:
+    The semantic version string, like "1.2.3".
+  """
+  for i in range(len(bazel_version)):
+    c = bazel_version[i]
+    if not (c.isdigit() or c == "."):
+      return bazel_version[:i]
+  return bazel_version
+
+
+def _parse_bazel_version(bazel_version):
+  """Parses a version string into a 3-tuple of ints
+  int tuples can be compared directly using binary operators (<, >).
+  Args:
+    bazel_version: the Bazel version string
+  Returns:
+    An int 3-tuple of a (major, minor, patch) version.
+  """
+
+  version = _extract_version_number(bazel_version)
+  return tuple([int(n) for n in version.split(".")])
+
+
+def do_preparation():
+  """Prepartion work before starting the migration."""
+  exit_code, stdout, _ = execute_command(["bazel", "--version"])
+  eprint(stdout.strip())
+  if exit_code != 0 or not stdout:
+    warning("Current bazel is not a release version.")
+    eprint("Please make sure you are running at least bazel 5.1.0")
+  elif _parse_bazel_version(stdout.strip().split(" ")[1]) < (5, 1, 0):
+    error("Current Bazel version  is older than 5.1.0")
+    eprint("Please make sure you are running at least bazel 5.1.0")
+    abort_migration()
+
+
+  # Create MODULE.bazel file if it doesn't exist already.
+  scratch_file("MODULE.bazel", [], mode="a")
```

Comment:
maybe create these with the current workspace name? (or do you think it's better to let the user manually add a module name later on?)

Reply:
I think let the user be aware of this is better.